File: Jessica/53.Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        MaxSum=nums[0]
        for i in range(1,len(nums)):
            if nums[i-1]>0:
                nums[i]+=nums[i-1]
            else:
                nums[i]+=0
            if nums[i]>MaxSum:
                MaxSum=nums[i]
        return MaxSum
                
            
        

File: Jessica/53.Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int Sum=0;
        int Max_Num=nums[0];
        for(int i=0;i<nums.size();i++){
            Sum+=nums[i];
            Max_Num=max(Max_Num,Sum);
            Sum=max(Sum,0);
        }
        return Max_Num;
    }
};
